flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/serializer/event/DataChangeEventSerializer.java

```diff
@@ -76,28 +72,18 @@ public DataChangeEvent deserialize(DataInputView source) throws IOException {
         OperationType op = opSerializer.deserialize(source);
         TableId tableId = tableIdSerializer.deserialize(source);
 
+        RecordData before = recordDataSerializer.deserialize(source);
+        RecordData after = recordDataSerializer.deserialize(source);
+        Map<String, String> meta = metaSerializer.deserialize(source);
```

Comment:
Is it safe to deserialize here if `op` isn't valid? Or we can validate op in advance.

Reply:
@yuxiqian Thanks for your review.

> Is it safe to deserialize here if `op` isn't any known following one? 

IMO, the currently deserialized fields and order are consistent with those during serialization. Even if an unknown op operation occurs, deserialization will not cause an exception.